src/DoctrineListener/DoctrineAuditListener.php

```diff
@@ -0,0 +1,253 @@
+                    }
+                }
+
+                break;
+            case EventAction::CREATE:
+                $entityData = $this->serializer->normalize($entity);
+                $entityFields = array_keys($entityData);
+                $trailFields = [];
+
+                // Hide sensitive data
+                $reflectionClass = new ReflectionClass($entity);
+                $properties = $reflectionClass->getProperties();
+                foreach ($properties as $property) {
+                    if ($property->getAttributes(AuditTrailSensitiveData::class)) {
+                        $property->setAccessible(true);
+                        $entityData[$property->getName()] = '*****';
+                    }
+
+                    // If the attribute AuditTrailDisplayData::class isn't set, we don't want to log it
+                    if ($property->getAttributes(AuditTrailDisplayData::class)) {
+                        $trailFields[] = $property->getName();
+                    }
+                }
+
+                // Remove all fields that aren't in the trailFields array
+                foreach ($entityData as $field => $value) {
+                    if (!in_array($field, $trailFields)) {
+                        unset($entityData[$field]);
+                    }
+                }
+
+                break;
+            default:
+                $sensitiveFields = [];
+                $trailFields = [];
+                $reflectionClass = new ReflectionClass($entity);
+                $properties = $reflectionClass->getProperties();
+                foreach ($properties as $property) {
+                    if ($property->getAttributes(AuditTrailSensitiveData::class)) {
+                        $property->setAccessible(true);
+                        $sensitiveFields[] = $property->getName();
+                    }
+
+                    // If the attribute AuditTrailDisplayData::class isn't set, we don't want to log it
+                    if ($property->getAttributes(AuditTrailDisplayData::class)) {
+                        $trailFields[] = $property->getName();
+                    }
+                }
+
+                $uow = $entityManager->getUnitOfWork();
+                $entityData = $uow->getEntityChangeSet($entity);
+                $entityFields = array_keys($entityData);
+                foreach ($entityData as $field => $change) {
+                    if (!in_array($field, $trailFields)) {
+                        unset($entityData[$field]);
+
+                        continue;
+                    }
+
+                    if (in_array($field, $sensitiveFields)) {
+                        $change = ['*****', '*****'];
+                    }
+
+                    $entityData[$field] = [
+                        'from' => $change[0],
+                        'to' => $change[1],
+                    ];
+                }
+        }
+
+        $this->auditLogger->log(
```

Comment:
Een entity wordt nog altijd gelogd, ook al zijn er geen properties die gelogd worden?

Reply:
Er worden altijd fields gelogged, maar de data is leeg by default. Je weet dus welke velden zijn aangepast maar niet naar welke value